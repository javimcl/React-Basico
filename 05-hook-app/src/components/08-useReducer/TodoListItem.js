import React from 'react';

export const TodoListItem = ({todo, i, handleToggle, handleDelete }) => {
    return (
        <>
            <li key={todo.id}
                className="list-group-item"
            >
                <p className={`${todo.done && 'complete'}`}
                    onClick={() => handleToggle(todo.id)}>
                    {i + 1}. {todo.desc}
                </p>
                <button

                    onClick={() => handleDelete(todo.id)}
                    className='btn btn-danger'>
                    Borrar
                </button>
            </li>
        </>
    );
};
