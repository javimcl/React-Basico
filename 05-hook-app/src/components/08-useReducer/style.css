body {
    padding: 70px;
}

p {
    cursor: pointer;
    margin: opx;

}

.list-group-item {
    align-items: center;
    display: flex;
    justify-content: space-between;
}

.complete {
    text-decoration: line-through;
}