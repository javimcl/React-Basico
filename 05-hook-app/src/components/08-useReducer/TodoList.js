import React from 'react';
import { TodoListItem } from './TodoListItem';

export const TodoList = ({todos, handleDelete, handleToggle}) => {
    return (
        <div>
            
            <ui className="list-group list-group-flush">
                        {
                            todos.map((todo, i) => (
                                <TodoListItem key={todo.id} todo={todo} i={i} handleDelete={handleDelete} handleToggle={handleToggle} />
                            ))
                        }
                    </ui>
            
        </div>
    );
};
