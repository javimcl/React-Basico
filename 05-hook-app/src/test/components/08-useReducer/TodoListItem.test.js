import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';


import { TodoListItem } from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodo";


describe('Pruebas en <TodoListItem/>', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(
        <TodoListItem
            todo={demoTodos[1]}
            i={0}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
        />
    )

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de llamar la funcion delete', () => {
        //jest.fn()
        //toHaveBeenCalledWith
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[1].id);
    });

    test('debe de llamar la funcion Toggle', () => {
        //jest.fn()
        //toHaveBeenCalledWith
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[1].id);
    });

    test('debe de mostrar el texto correctamente', () => {
        //contenido del parrafo
        //expect(wrapper.find('p').text()).toBe('1. ' + demoTodos[1].desc)
        expect(wrapper.find('p').text()).toBe(`1. ${demoTodos[1].desc}`)
    });

    test('debe de tener la clase complete', () => {
       
        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow(
            <TodoListItem
                todo={todo}
               
            />
        );

        expect(wrapper.find('p').hasClass('complete')).toBe(true);


    });








});
