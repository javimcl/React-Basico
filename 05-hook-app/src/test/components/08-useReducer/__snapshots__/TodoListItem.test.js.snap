// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="2"
  >
    <p
      className="false"
      onClick={[Function]}
    >
      1
      . 
      Aprender Mongo
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Borrar
    </button>
  </li>
</Fragment>
`;
